src/transformers/modeling_tf_utils.py

```diff
@@ -428,7 +428,14 @@ def run_call_with_unpacked_inputs(self, *args, **kwargs):
         # process the inputs and call the wrapped function
         main_input_name = getattr(self, "main_input_name", func.__code__.co_varnames[1])
         main_input = fn_args_and_kwargs.pop(main_input_name, None)
-        unpacked_inputs = input_processing(func, self.config, main_input, **fn_args_and_kwargs)
+
+        # Encoder Decoder models delegate the application of the configuration options to their inner models.
+        if "encoder_decoder" in str(self).lower():
```

Comment:
Just a nit, no need to feel strong:

I would personally prefer to use `self.__name__` instead of `str` and test against `EncoderDecoder`. If I understand correctly, `str` gives the full path, which includes the module name like `encoder_decoder` or `vision_encoder_decoder`, right?

Reply:
I was trying `self.__name__` on this line, and it seems like it isn't defined in some cases -- e.g. in `tests/models/vision_encoder_decoder/test_modeling_tf_vision_encoder_decoder.py::TFViT2GPT2EncoderDecoderModelTest::test_encoder_decoder_model`, it throws `*** AttributeError: 'TFVisionEncoderDecoderModel' object has no attribute '__name__'`

`str(self)` here includes the full import path for the object, like you wrote, which contains the class name -- `'<transformers.models.vision_encoder_decoder.modeling_tf_vision_encoder_decoder.TFVisionEncoderDecoderModel object at 0x7fc7783b8a60>'`

Reply:
Maybe we can update in the future, but since it is working for now I'm going with it :D

Reply:
OK, thanks for the info.